space_robots/Dockerfile

```diff
@@ -32,42 +32,45 @@ LABEL org.label-schema.description="Curiosity rover demo on the Space ROS platfo
 LABEL org.label-schema.vendor="Open Robotics"
 LABEL org.label-schema.version=${VERSION}
 LABEL org.label-schema.url="https://github.com/space-ros"
-LABEL org.label-schema.vcs-url="https://github.com/space-ros/docker-images"
+LABEL org.label-schema.vcs-url="https://github.com/space-ros/docker"
 LABEL org.label-schema.vcs-ref=${VCS_REF}
 
-# Disable prompting during package installation
-ARG DEBIAN_FRONTEND=noninteractive
-
 # Define a few key variables
-ARG SPACEROS_DEMO_DIR=${SRC_DIR}/spaceros_demo
+ARG DEMO_DIR=${HOME_DIR}/demos_ws
 ENV IGNITION_VERSION fortress
+ENV GZ_VERSION fortress
+
+# Disable prompting during package installation
+ARG DEBIAN_FRONTEND=noninteractive
+RUN /bin/bash -c 'echo 'debconf debconf/frontend select Noninteractive' | sudo debconf-set-selections' \
+&& sudo apt-get install -y -q
 
 # Get rosinstall_generator
 RUN sudo apt-get update -y && sudo apt-get install -y python3-rosinstall-generator
 
 # Generate repos file for demo dependencies, excluding packages from Space ROS core.
-RUN mkdir -p ${SPACEROS_DEMO_DIR}/src
-WORKDIR ${SPACEROS_DEMO_DIR}
-COPY --chown=${USERNAME}:${USERNAME} spaceros-demo-pkgs.txt /tmp/
-COPY --chown=${USERNAME}:${USERNAME} excluded-pkgs.txt /tmp/
+COPY --chown=${USERNAME}:${USERNAME} demo-pkgs.txt /tmp/
 RUN rosinstall_generator \
   --rosdistro ${ROSDISTRO} \
   --deps \
   --exclude-path ${SPACEROS_DIR}/src \
-  --exclude $(cat /tmp/excluded-pkgs.txt) -- \
-  $(cat /tmp/spaceros-demo-pkgs.txt) \
-  > /tmp/spaceros_demo_generated_pkgs.repos
+  -- $(cat /tmp/demo-pkgs.txt) \
+  > /tmp/demo_generated_pkgs.repos
+
+RUN mkdir -p ${DEMO_DIR}/src
+WORKDIR ${DEMO_DIR}
 
 # Get the source for the dependencies
-RUN vcs import src < /tmp/spaceros_demo_generated_pkgs.repos
-COPY --chown=${USERNAME}:${USERNAME} spaceros_demo_manual_pkgs.repos /tmp/
-RUN vcs import src < /tmp/spaceros_demo_manual_pkgs.repos --force
+RUN vcs import src < /tmp/demo_generated_pkgs.repos
+COPY --chown=${USERNAME}:${USERNAME} demo_manual_pkgs.repos /tmp/
+RUN vcs import src < /tmp/demo_manual_pkgs.repos --force
 
 # Get a version of gz_ros2_control to avoid a gazebo crash (still investigating the root cause)
-RUN cd src/gz_ros2_control/ign_ros2_control && git checkout 48ba51afb20e5a75704e6315f2cfe54b1b572ff4
+RUN cd ${DEMO_DIR}/src/gz_ros2_control/ign_ros2_control && git checkout 48ba51afb20e5a75704e6315f2cfe54b1b572ff4
```

Comment:
Rather than doing this in a separate step, this checkout of ign_ros2_control should be put in the demo_manual_pkgs.repos file with this commit specified as the version.

I know this was here before this PR but I think it should be changed now while we're revisiting it.

Reply:
Seems to be working with the humble branch now